src/Streamly/Prelude.hs

```diff
@@ -120,6 +123,20 @@ unfoldrM step = fromStream . go
 each :: (Streaming t, Foldable f) => f a -> t m a
 each = Prelude.foldr cons nil
 
+-- | Iterate a pure function from a seed value, streaming the results forever
+iterate :: Streaming t => (a -> a) -> a -> t m a
+iterate step = fromStream . go
+  where
```

Comment:
Use 4 space indentation everywhere. The `fromHandle` function below is an aberration. The rest of the file is using 4 spaces.